Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,426 @@
+/**
+ * @copyright Copyright (c) 2022 UT Longhorn Racing Solar
+ * 
+ * @file SendTritium.c
+ * @brief Function implementations for the SendTritium application.
+ * 
+ * This contains functions relevant to updating the velocity and current setpoitns
+ * of the Tritium motor controller. The implementation includes a normal current
+ * controlled mode, a one-pedal driving mode, cruise control, and regenerative braking.
+ * The logic is determined through a finite state machine implementation.
+ * 
+ * @author Nathaniel Delgado (NathanielDelgado)
+ * @author Diya Rajon (diyarajon)
+ * @author Ishan Deshpande (IshDeshpa)
+ */
+
+#include "Pedals.h"
+#include "ReadCarCAN.h"
+#include "Minions.h"
+#include "SendTritium.h"
+#include "ReadTritium.h"
+#include "CANbus.h"
+
+
+// Macros
+#define MAX_VELOCITY 50.0f // m/s    if rmp: 20000.0f // rpm
+#define NORMAL_BRAKE_THRESHOLD 5  // percent
+#define ONEPEDAL_BRAKE_THRESHOLD 20  // percent
+#define NEUTRAL_THRESHOLD 25    // percent
+#define DEBOUNCE_PERIOD 2 // in 100 ms
+#define VELOCITY_THRESHOLD 20.0f // m/s
+
+#define MOTOR_MSG_PERIOD 100
+#define FSM_PERIOD 20
+#define MOTOR_MSG_COUNTER_THRESHOLD (MOTOR_MSG_PERIOD)/(FSM_PERIOD)
+
+#ifdef __TEST_SENDTRITIUM
+#define SCOPE
+#else
+#define SCOPE static
+static int MotorCounter = 0;
+#endif
+
+// Inputs
+SCOPE bool cruiseEnable = false;
+SCOPE bool cruiseSet = false;
+SCOPE bool regenToggle = false;
+SCOPE uint8_t brakePedalPercent = 0;
+SCOPE uint8_t accelPedalPercent = 0;
+SCOPE bool reverseSwitch = false;
+SCOPE bool forwardSwitch = false;
+
+// Outputs
+SCOPE float currentSetpoint = 0;
+SCOPE float velocitySetpoint = 0;
+SCOPE float cruiseVelSetpoint = 0;
+static bool brakelightState = false;
```

Comment:
Since the printing is now self contained in SendTritium, we need to reevaluate which of these inputs/outputs actually need to be non-static (if we don't use the variable in the test file, we don't need to expose the variable, so it can stay static all the time and we can remove the SCOPE macro and just replace it with static).